airbyte-integrations/connectors/source-public-apis/source_public_apis/components.py

```diff
@@ -0,0 +1,15 @@
+from typing import Any, Iterable, Mapping
+from airbyte_cdk.sources.streams.http import HttpStream
+from airbyte_cdk.sources.declarative.extractors.record_extractor import RecordExtractor
+from typing import Any, List
+import requests
+
+class CustomExtractor(RecordExtractor):
+    
+    def extract_records(self, response: requests.Response, **kwargs) -> List[Mapping[str, Any]]:
+
+        extracted=[]
+        print(response.json())
+        for cat in response.json()['categories']:
+            extracted.append({"name": cat})
+        return extracted
```

Comment:
Can you keep the same function as before `        return [{"name": cat} for cat in response.json()["categories"]]
`

Reply:
resolved